source/particle/interpolator/bilinear_least_squares.cc

```diff
@@ -0,0 +1,162 @@
+/*
+  Copyright (C) 2016 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#include <aspect/particle/interpolator/bilinear_least_squares.h>
+
+#include <deal.II/grid/grid_tools.h>
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Interpolator
+    {
+      /**
+       * Return the cell-wise evaluated properties of the bilinear least squares function at the positions.
+       */
+      template <int dim>
+      std::vector<std::vector<double> >
+      BilinearLeastSquares<dim>::properties_at_points(const std::multimap<types::LevelInd, Particle<dim> > &particles,
+                                                      const std::vector<Point<dim> > &positions,
+                                                      const typename parallel::distributed::Triangulation<dim>::active_cell_iterator &cell) const
+      {
+        AssertThrow(dim == 2,
```

Comment:
For now, I'll have only implemented the 2D case. The 3D case is in the works and will be added on soon.